.travis.yml

```diff
@@ -114,6 +106,16 @@ cache:
 
 before_install:
   - ccache -M 2G && ccache -z
+  # Install kcov manually, since it's not packaged for bionic beaver
+  - if [[ "$KCOV_VERSION" ]]; then
+        sudo apt-get -y install cmake binutils-dev libcurl4-openssl-dev zlib1g-dev libdw-dev libiberty-dev;
+        KCOV_ROOT=$(mktemp -d);
+        wget --output-document=${KCOV_ROOT}/${KCOV_VERSION}.tar.gz https://github.com/SimonKagstrom/kcov/archive/v${KCOV_VERSION}.tar.gz;
+        tar -C ${KCOV_ROOT} -xzvf ${KCOV_ROOT}/${KCOV_VERSION}.tar.gz;
+        mkdir -p ${KCOV_ROOT}/build;
+        cd ${KCOV_ROOT}/build && cmake -Wno-dev ${KCOV_ROOT}/kcov-${KCOV_VERSION} && cd - ;
+        make -C ${KCOV_ROOT}/build && sudo  make -C ${KCOV_ROOT}/build install;
```

Comment:
Can we use spack for this? We should if possible. In general I think we should prefer to dogfood even in tests, unless there’s a reason not to.

Reply:
If we go all the way down this means installing:
```
kcov@38%gcc@9.0.1 build_type=RelWithDebInfo arch=linux-ubuntu18.04-broadwell
    ^cmake@3.17.1%gcc@9.0.1~doc+ncurses+openssl+ownlibs patches=1c540040c7e203dd8e27aa20345ecb07fe06570d56410a24a266ae570b1c4c39 ~qt arch=linux-ubuntu18.04-broadwell
        ^ncurses@6.2%gcc@9.0.1~symlinks+termlib arch=linux-ubuntu18.04-broadwell
            ^pkgconf@1.6.3%gcc@9.0.1 arch=linux-ubuntu18.04-broadwell
        ^openssl@1.1.1g%gcc@9.0.1+systemcerts arch=linux-ubuntu18.04-broadwell
            ^perl@5.30.2%gcc@9.0.1+cpanm+shared+threads arch=linux-ubuntu18.04-broadwell
                ^gdbm@1.18.1%gcc@9.0.1 arch=linux-ubuntu18.04-broadwell
                    ^readline@8.0%gcc@9.0.1 arch=linux-ubuntu18.04-broadwell
            ^zlib@1.2.11%gcc@9.0.1+optimize+pic+shared arch=linux-ubuntu18.04-broadwell
    ^curl@7.68.0%gcc@9.0.1~darwinssl~gssapi~libssh~libssh2~nghttp2 arch=linux-ubuntu18.04-broadwell
```
which may take a long time under CI, unless we want to register most of the things as externals but that would complicate a bit the setup and defeat the idea of dogfooding. If we only had binary packages... :slightly_smiling_face:

Reply:
yeah ok let's hold off until we have binaries.  Gosh that'll be nice.